packages/backend-function/src/function_env_type_generator.ts

```diff
@@ -21,10 +22,33 @@ export class FunctionEnvironmentTypeGenerator {
    * Initialize typed process.env shim file name and location
    */
   constructor(private readonly functionName: string) {
-    this.typeDefFilePath = `${process.cwd()}/.amplify/generated/env/${
-      this.functionName
-    }.ts`;
+    this.typeDefFilePath = path.join(
+      process.cwd(),
+      '.amplify',
+      'generated',
+      'env',
+      `${this.functionName}.ts`
+    );
+    this.clearGeneratedEnvDirectory();
   }
+  /**
+   * Clear existing files and subdirectories in the generated env directory
+   */
+  public clearGeneratedEnvDirectory = (): void => {
```

Comment:
can this be private ?

Reply:
Can't be private since we're using this in test class